File: C++/VendingMachine/VendingMachine/RailConfigurationSettings.cpp
#include "pch.h"
#include "RailConfigurationSettings.h"
#include <utility>

RailConfigurationSettings::RailConfigurationSettings(
	std::string railCode, const double price, const unsigned int initialInventory, std::string label)
	: RailCode(std::move(railCode)), Price(price), InitialInventory(initialInventory), Label(std::move(label))
{
}

std::string RailConfigurationSettings::GetRailCode() const
{
	return RailCode;
}

double RailConfigurationSettings::GetPrice() const
{
	return Price;
}

unsigned int RailConfigurationSettings::GetInitialInventory() const
{
	return InitialInventory;
}

std::string RailConfigurationSettings::GetLabel() const
{
	return Label;
}

File: java/VendingMachine/src/com/ravendarque/vendingMachine/rails/RailConfigurationSettings.java
package com.ravendarque.vendingMachine.rails;

public class RailConfigurationSettings {

    private final String railCode;
    private final double price;
    private final int initialInventory;
    private final String label;

    public RailConfigurationSettings(
            String railCode, double price, int initialInventory, String label) {

        this.railCode = railCode;
        this.price = price;
        this.initialInventory = initialInventory;
        this.label = label;
    }

    String getLabel() {

        return label;
    }

    int getInitialInventory() {

        return initialInventory;
    }

    String getRailCode() {

        return railCode;
    }

    double getPrice() {

        return price;
    }

}
